oasst-shared/oasst_shared/schemas/inference.py

```diff
@@ -117,6 +117,82 @@ class SamplingParameters(pydantic.BaseModel):
     max_new_tokens: int = 1024
 
 
+class PluginApiType(pydantic.BaseModel):
+    type: str
+    url: str
+    has_user_authentication: bool | None = False
+    # NOTE: Some plugins using this field,
+    # instead of has_user_authentication
+    is_user_authenticated: bool | None = False
+
+
+class PluginAuthType(pydantic.BaseModel):
+    type: str
+
+
+class PluginOpenAPIParameter(pydantic.BaseModel):
+    name: str
+    in_: str
+    description: str
+    required: bool
+    schema_: object
+
+
+class PluginOpenAPIEndpoint(pydantic.BaseModel):
+    path: str
+    type: str
+    summary: str
+    operation_id: str
+    url: str
+    params: List[PluginOpenAPIParameter]
+
+
+class PluginConfig(pydantic.BaseModel):
+    schema_version: str
+    name_for_model: str
+    name_for_human: str
+    description_for_human: str
+    description_for_model: str
+    api: PluginApiType
+    auth: PluginAuthType
+    logo_url: str | None = None
+    contact_email: str | None = None
+    legal_info_url: str | None = None
+    endpoints: List[PluginOpenAPIEndpoint] | None = None
+
+    def __getitem__(self, key: str) -> Any:
```

Comment:
are you sure these are working as wanted? pydantic does quite a bit of magic, I'm particularly worried about the use of `setattr` here, but it's just a feeling, no particular knowledge

Reply:
I am not sure, It works. but could be that I am misusing it?